test/erc20_gateway/erc20_cogateway/withdraw.js

```diff
@@ -0,0 +1,163 @@
+      web3.eth.abi.encodeParameters(
+        ['bytes32', 'address', 'address', 'uint256', 'address'],
+        [ 
+          WITHDRAW_INTENT_TYPEHASH, 
+          valueToken,
+          utilityToken,
+          amount.toString(), 
+          beneficiary],
+      ),
+    );
+  }
+
+  beforeEach(async () => {
+    erc20Cogateway = await ERC20Cogateway.new();
+    const owner = accountProvider.get();
+    sender = accountProvider.get();
+    utilityToken = await Utils.deployMockToken(owner);
+
+    const valueToken = '0x0000000000000000000000000000000000000000';
+
+    setupGenesisParams = {
+      genesisMetachainId: Utils.generateRandomMetachainId(),
+      genesisERC20Gateway: accountProvider.get(),
+      genesisStateRootProvider: accountProvider.get(),
+      genesisMaxStorageRootItems: new BN(100),
+      genesisOutboxStorageIndex: new BN(4),
+    };
+    
+    withdrawParam = {
+      utilityToken: utilityToken.address,
+      amount: new BN(100),
+      beneficiary: accountProvider.get(),
+      feeGasPrice: new BN(1),
+      feeGasLimit: new BN(1),
+    };
+
+    await utilityToken.transfer(
+      sender,
+      withdrawParam.amount,
+      { from: owner },
+    );
+
+    await erc20Cogateway.setupGenesis(
+      setupGenesisParams.genesisMetachainId,
+      setupGenesisParams.genesisERC20Gateway,
+      setupGenesisParams.genesisStateRootProvider,
+      setupGenesisParams.genesisMaxStorageRootItems,
+      setupGenesisParams.genesisOutboxStorageIndex,
+    );
+
+    await erc20Cogateway.setup();
+
+    outboundChannelIdentifier = await erc20Cogateway.outboundChannelIdentifier.call();
+
+    withdrawIntent = await hashWithdrawIntent(
+      valueToken,
+      withdrawParam.utilityToken,
+      withdrawParam.amount,
+      withdrawParam.beneficiary,
+    );
+
+  });
+
+  it('should successfully withdraw' , async () => {
+    const beforeSenderUtilityTokenBalance = await utilityToken.balanceOf(sender);
+
+    await utilityToken.approve(erc20Cogateway.address, withdrawParam.amount, {
+      from: sender,
+    });
+
+    const calculatedMessageHash = await MessageBusUtils.hashMessage(
+      withdrawIntent,
+      new BN(0),
+      withdrawParam.feeGasPrice,
+      withdrawParam.feeGasLimit,
+      sender,
+      outboundChannelIdentifier,
+    );
+
+    const messageHash = await erc20Cogateway.withdraw.call(
+      withdrawParam.utilityToken,
+      withdrawParam.amount,
+      withdrawParam.beneficiary,
+      withdrawParam.feeGasPrice,
+      withdrawParam.feeGasLimit,
+      { from: sender },
+    );
+
+    await erc20Cogateway.withdraw(
+      withdrawParam.utilityToken,
+      withdrawParam.amount,
+      withdrawParam.beneficiary,
+      withdrawParam.feeGasPrice,
+      withdrawParam.feeGasLimit,
+      { from: sender },
+    );
+
+    const afterSenderUtilityTokenBalance = await utilityToken.balanceOf(sender);
+
+    assert.strictEqual(
+      messageHash,
+      calculatedMessageHash,
+      "Incorrect Message hash."
```

Comment:
```suggestion
      'Incorrect Message hash.'
```
Strings must use singlequote.

Reply:
Fixed.